### PR TITLE
Maven fixes

### PR DESCRIPTION
**Pull request by cristianonicolai**

### CODE REVIEW

File: kogito-quarkus-extension/bom/pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
   <parent>
     <groupId>org.kie.kogito</groupId>
-    <artifactId>kogito-bom</artifactId>
+    <artifactId>kogito-quarkus-parent</artifactId>
```
Comment (evacchi) [inline kogito-quarkus-extension/bom/pom.xml]: I think this was intentional. @mariofusco ?

Comment (mariofusco) [reply kogito-quarkus-extension/bom/pom.xml]: @evacchi nope, I think this fix is more correct. Not sure how this was breaking anything though.

File: kogito-quarkus-extension/runtime/pom.xml [hunk 2]
```diff
@@ -74,9 +74,7 @@
             <plugin>
                 <groupId>io.quarkus</groupId>
                 <artifactId>quarkus-bootstrap-maven-plugin</artifactId>
-            </plugin>
-            <plugin>
-                <artifactId>maven-compiler-plugin</artifactId>
+                <version>${version.io.quarkus}</version>
```
Comment (MarianMacik) [inline kogito-quarkus-extension/runtime/pom.xml]: What about specifying this plugin in the `pluginManagement` section in `kogito-runtimes` parent pom.xml alongside `quarkus-maven-plugin`?

https://github.com/kiegroup/kogito-runtimes/blob/4ba3fc39c2f6d93a7f1cc75ee98ed657c90634e3/pom.xml#L1243-L1247

Comment (cristianonicolai) [reply kogito-quarkus-extension/runtime/pom.xml]: +1, done

**Pull request merged**